specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json

```diff
@@ -3658,6 +3807,40 @@
         "message"
       ]
     },
+    "TriggerSubscriptionOperationStatus": {
+      "description": "Defines the response of a trigger subscription operation.",
+      "type": "object",
+      "properties": {
+        "triggerName": {
+          "description": "Trigger name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "status": {
+          "description": "Subscription status.",
+          "$ref": "#/definitions/EventSubscriptionStatusTypes",
+          "readOnly": true
+        }
+      }
+    },
+    "EventSubscriptionStatusTypes": {
```

Comment:
You don't need to declare it a new type. The types make sense when they are used or can be used in future in multiple APIs.  It doesn't seem to be the case here. Also the type you have here is array - can you returned object have multiple statuses?
You should just do something similar to this TriggerRunStatus for your status property. And remove types from the name.
       "status": {
          "type": "string",
          "enum": [
            "Succeeded",
            "Failed",
            "Inprogress"
          ],
          "x-ms-enum": {
            "name": "TriggerRunStatus",
            "modelAsString": true
          },
          "description": "Trigger run status.",
          "readOnly": true